### PR TITLE
Remove `Data` field from serialized Transactions

### PR DESCRIPTION
**Pull request by emlautarom1**
Partially addresses #5161

## Changes

- Do not serialize the `Data` field, but use it as alias of `Input` when deserializing.

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [x] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Notes on testing

- Test suite has been updated to reflect the changes
- Tested syncing Sepolia with Prysm v4.0.8

#### Requires documentation update

- [x] Yes
- [ ] No

#### Requires explanation in Release Notes

- [x] Yes
- [ ] No

## Remarks

While working on #5161 we found that the `Data` field is not part of the spec (https://ethereum.github.io/execution-apis/api-documentation/), so we removed it on #6036.

As documented on that PR, this could introduce issues with users that relied on this non-conformant field. During internal testing, we found that the Prysm CL relied on this so we rolled back the change on #6067.

After discussing it with the team, we decided to compromise in the following fashion: we will accept `Data` as an alias for `Input` when reading transactions from other applications, but we will not include such field when sending them.

### CODE REVIEW

File: src/Nethermind/Nethermind.JsonRpc/Data/TransactionForRpc.cs [hunk 1]
```diff
@@ -91,7 +91,12 @@ public TransactionForRpc()
 
     public UInt256? MaxFeePerGas { get; set; }
     public long? Gas { get; set; }
-    public byte[]? Data { get; set; }
+
+    // Required for compatibility with some CLs like Prysm
+    // Accept during deserialization, ignore during serialization
+    // See: https://github.com/NethermindEth/nethermind/pull/6067
+    [JsonProperty(nameof(Data))]
+    private byte[]? Data { set { Input = value; } }
```
Comment (emlautarom1) [inline src/Nethermind/Nethermind.JsonRpc/Data/TransactionForRpc.cs:99]: The "trick" used to support deserialization but ignore serialization is described here: https://stackoverflow.com/questions/43714050/multiple-jsonproperty-name-assigned-to-single-property/43715009#43715009

**Review by LukaszRozmej — APPROVED** [review]

**Review by smartprogrammer93 — APPROVED** [review]
I like this way of doing it

**Comment by LukaszRozmej** [comment]
Context: https://github.com/ethereum/go-ethereum/issues/28077

**Review by benaadams — APPROVED** [review]
Would be good for me if this was merged sooner rather than later; as will be additional merge conflicts 😅

**Comment by smartprogrammer93** [comment]
Let me just run it with prysm to make sure it does not break again then ill merge

**Pull request merged**